libpod/container_internal_common.go

```diff
@@ -679,6 +699,40 @@ func (c *Container) generateSpec(ctx context.Context) (s *spec.Spec, cleanupFunc
 		}
 	}
 
+	// There's no straight-forward way to copy symlinks around, and we need
+	// to be extra careful to _always_ operate behind the container's
+	// mountpoint and _never_ mess on the host.
+	//
+	// Hence, "preserving symlinks" boils down to creating a new symlink
+	// behind the container's mountpoint.  To mimic mount behavior, we need
+	// to create all directories up to the symlinks "mount destination".
+	if c.ensureState(define.ContainerStateConfigured, define.ContainerStateCreated) {
+		for source, target := range createSymlinks {
+			err := func() error {
+				resolvedSource, err := securejoin.SecureJoin(c.state.Mountpoint, source)
+				if err != nil {
+					return err
+				}
+				if err := os.Remove(resolvedSource); err != nil {
+					if !errors.Is(err, os.ErrNotExist) {
+						return err
+					}
+					resolvedDir, err := securejoin.SecureJoin(c.state.Mountpoint, filepath.Dir(source))
+					if err != nil {
+						return err
+					}
+					if err := os.MkdirAll(resolvedDir, 0755); err != nil {
+						return err
+					}
+				}
+				return os.Symlink(target, resolvedSource)
+			}()
```

Comment:
we need to handle it in a similar way to `Container.safeMountSubPath()` otherwise if the path is on a volume controlled by a running container, we might be tricked into creating the symlink on the host